src/overlay/Tracker.cpp

```diff
@@ -89,61 +90,92 @@ Tracker::tryNextPeer()
 {
     // will be called by some timer or when we get a
     // response saying they don't have it
-    Peer::pointer peer;
-
     CLOG(TRACE, "Overlay") << "tryNextPeer " << hexAbbrev(mItemHash)
                            << " last: "
                            << (mLastAskedPeer ? mLastAskedPeer->toString()
                                               : "<none>");
 
-    // if we don't have a list of peers to ask and we're not
-    // currently asking peers, build a new list
-    if (mPeersToAsk.empty() && !mLastAskedPeer)
+    if (mLastAskedPeer)
     {
-        std::set<std::shared_ptr<Peer>> peersWithEnvelope;
-        for (auto const& e : mWaitingEnvelopes)
-        {
-            auto const& s = mApp.getOverlayManager().getPeersKnows(e.first);
-            peersWithEnvelope.insert(s.begin(), s.end());
-        }
+        mTryNextPeer.Mark();
+        mLastAskedPeer.reset();
+    }
 
-        // move the peers that have the envelope to the back,
-        // to be processed first
-        for (auto const& p :
-             mApp.getOverlayManager().getRandomAuthenticatedPeers())
+    // helper function to populate "candidates" with the lowest latency group
+    std::vector<Peer::pointer> candidates;
+    int64 curBest = INT64_MAX;
+    auto procPeers = [&](std::map<NodeID, Peer::pointer> const& peerMap) {
+        for (auto& mp : peerMap)
         {
-            if (peersWithEnvelope.find(p) != peersWithEnvelope.end())
+            auto& p = mp.second;
+            if (p->isAuthenticated() &&
+                mPeersAsked.find(p) == mPeersAsked.end())
             {
-                mPeersToAsk.emplace_back(p);
+                constexpr int64 GROUPSIZE_MS = 500;
+                int64 plat = p->getPing().count() / GROUPSIZE_MS;
+                if (plat < curBest)
+                {
+                    candidates.clear();
+                    curBest = plat;
+                    candidates.emplace_back(p);
+                }
+                else if (curBest == plat)
+                {
+                    candidates.emplace_back(p);
+                }
             }
-            else
+        }
+    };
+
+    // build the set of peers we didn't ask yet that have this envelope
```

Comment:
Would appreciate a little more of a comment here: note that we're doing a somewhat arbitrary grouping of peers by latency-range, and taking the lowest-latency group. Also like .. why this is a good idea? It makes some sense but I can picture other options (eg. taking the lowest-latency and then `while(rand_flip()) { randomly replacing with next-lowest  }` or something).

Reply:
I am not sure what you're advocating for in terms of policy. Are you saying we could ignore latency with some probability (so mixing this and master with some probability?)?
Is it that you're trying to preserve some property from the "pure random" selection?

Reply:
I literally pulled that policy out of thin air. I guess that's all I'm saying here: bucketing by 500ms groups is a bit arbitrary too. I figured some exploration of options might be good either in conversation or comments here or whatever.